s4e-backend/src/main/java/pl/cyfronet/s4e/controller/GroupController.java

```diff
@@ -0,0 +1,177 @@
+package pl.cyfronet.s4e.controller;
+
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import lombok.RequiredArgsConstructor;
+import lombok.val;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.i18n.LocaleContextHolder;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import pl.cyfronet.s4e.bean.Group;
+import pl.cyfronet.s4e.controller.response.GroupResponse;
+import pl.cyfronet.s4e.event.OnAddToGroupEvent;
+import pl.cyfronet.s4e.event.OnRemoveFromGroupEvent;
+import pl.cyfronet.s4e.ex.BadRequestException;
+import pl.cyfronet.s4e.ex.GroupCreationException;
+import pl.cyfronet.s4e.ex.GroupUpdateException;
+import pl.cyfronet.s4e.ex.NotFoundException;
+import pl.cyfronet.s4e.service.AppUserService;
+import pl.cyfronet.s4e.service.GroupService;
+import pl.cyfronet.s4e.service.InstitutionService;
+import pl.cyfronet.s4e.service.SlugService;
+
+import javax.validation.Valid;
+import java.util.stream.Collectors;
+
+import static pl.cyfronet.s4e.Constants.API_PREFIX_V1;
+
+@RestController
+@RequestMapping(API_PREFIX_V1)
+@RequiredArgsConstructor
+public class GroupController {
+    private final GroupService groupService;
+    private final AppUserService appUserService;
+    private final InstitutionService institutionService;
+    private final ApplicationEventPublisher eventPublisher;
+    private final SlugService slugService;
+
+    @ApiOperation("Create a new group")
+    @ApiResponses({
+            @ApiResponse(code = 200, message = "If group was created"),
+            @ApiResponse(code = 400, message = "Group not created"),
+            @ApiResponse(code = 404, message = "Institution not found")
+    })
+    @PostMapping("/institutions/{institution}/groups")
+    public ResponseEntity<?> create(@RequestBody @Valid String name,
```

Comment:
Shouldn't name be passed as a key value from backend? I.e. have some CreateGroupRequest class with property name?
Currently @Valid is noop as well, and you could wire up some validation to such request class.